pkgs/tools/nix/nixos-render-docs/src/tests/test_redirects.py

```diff
@@ -0,0 +1,184 @@
+                infile.write(content)
+
+        md = HTMLConverter("1.0.0", HTMLParameters("", [], [], 2, 2, 2, Path("")), {}, Redirects({"redirects-test-suite": ["index.html"]} | redirects, ''))
+        md.convert(Path(__file__).parent / 'index.md', Path(__file__).parent / 'index.html')
+
+        return md._redirects
+
+    def test_identifier_added(self):
+        """Test adding a new identifier to the source."""
+        before = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}"},
+            redirects={"foo": ["foo.html"]},
+        )
+        before.raise_errors()
+
+        intermediate = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}\n## Bar {#bar}"},
+            redirects={"foo": ["foo.html"]},
+        )
+        self.assertRaises(MissingRedirectError, lambda: intermediate.raise_errors(MissingRedirectError))
+
+        after = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}\n## Bar {#bar}"},
+            redirects={"foo": ["foo.html"], "bar": ["foo.html"]},
+        )
+        after.raise_errors()
+
+    def test_identifier_removed(self):
+        """Test removing an identifier from the source."""
+        before = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}\n## Bar {#bar}"},
+            redirects={"foo": ["foo.html"], "bar": ["foo.html"]},
+        )
+        before.raise_errors()
+
+        intermediate = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}"},
+            redirects={"foo": ["foo.html"], "bar": ["foo.html"]},
+        )
+        self.assertRaises(OrphanIdentifiersError, lambda: intermediate.raise_errors(OrphanIdentifiersError))
+
+        after = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}"},
+            redirects={"foo": ["foo.html"]},
+        )
+        after.raise_errors()
+
+    def test_identifier_renamed(self):
+        """Test renaming an identifier in the source."""
+        before = self.setup_boilerplate(
+            sources={"foo.md": "# Foo {#foo}\n## Bar {#bar}"},
+            redirects={"foo": ["foo.html"], "bar": ["foo.html"]},
+        )
+        before.raise_errors()
+
+        intermediate = self.setup_boilerplate(
+            sources={"foo.md": "# Foo Prime {#foo-prime}\n## Bar {#bar}"},
+            redirects={"foo": ["foo.html"], "bar": ["foo.html"]},
+        )
+        self.assertRaises(MissingRedirectError, lambda: intermediate.raise_errors(MissingRedirectError))
+        self.assertRaises(OrphanIdentifiersError, lambda: intermediate.raise_errors(OrphanIdentifiersError))
+
+        after = self.setup_boilerplate(
+            sources={"foo.md": "# Foo Prime {#foo-prime}\n## Bar {#bar}"},
+            redirects={"foo-prime": ["foo.html", "foo.html#foo"], "bar": ["foo.html"]},
+        )
+        after.raise_errors()
+
+    def test_leaf_identifier_moved_to_different_file(self):
+        """Test moving a leaf identifier to a different source file."""
```

Comment:
```suggestion
        """Test moving a leaf identifier to a different output file."""
```

This is done implicitly through how the boilerplate is set up, so it's not enough to say it's just moving to a different source file.